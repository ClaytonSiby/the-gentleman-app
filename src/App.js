import React from 'react';
import { Route, Router, Switch } from 'react-router-dom';
import { createBrowserHistory } from 'history';
import { Container } from 'react-bootstrap';
import Homepage from './pages/Homepage';
import Appointments from './components/Appointments';
import SignUp from './pages/SignUp';
import Login from './pages/Login';
import Suits from './components/Suits';
import HomepageLayout from './Layouts/HomepageLayout';
import MainLayout from './Layouts/MainLayout';
import styles from './assets/scss/app.module.scss';

const history = createBrowserHistory();

const App = () => (
  <Container fluid className={`${styles.app}`}>
    <Router history={history}>
      <Switch>
        <Route
          exact
          path="/"
          render={() => (
            <HomepageLayout>
              <Homepage />
            </HomepageLayout>
          )}
        />

        <Route
          path="/signup"
          render={() => (
            <MainLayout>
              <SignUp />
            </MainLayout>
          )}
        />

        <Route
          path="/login"
          render={() => (
            <MainLayout>
              <Login />
            </MainLayout>
          )}
        />

        <Route
          path="/suits"
          render={() => (
            <MainLayout>
              <Suits />
            </MainLayout>
          )}
        />

        <Route path="/appointments" render={() => (
          <MainLayout>
            <Appointments />
          </MainLayout>
        )}/>

        <Route path="/logout" render={() => {
          localStorage.setItem('userToken', '');
          window.location.href = "/login";
        }}/>
      </Switch>
    </Router>
  </Container>
);

export default App;
