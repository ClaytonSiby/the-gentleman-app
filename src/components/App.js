import React from 'react';
import styles from '../assets/scss/app.module.scss';

const App = () => (
  <div className={`${styles.app}`}>
    <h2>Hello World</h2>
    <p>I love programming</p>
  </div>
);

export default App;
