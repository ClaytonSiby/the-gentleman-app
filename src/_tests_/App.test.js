import React from 'react';
import { render } from '@testing-library/react';
import App from '../App';

describe('RouteGenerator', () => {
    it('Renders the RouteGenerator successfully', () => {
        render(<App />)
    })
})
