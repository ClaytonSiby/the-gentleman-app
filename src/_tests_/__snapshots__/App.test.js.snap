// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RouteGenerator renders the output consistently to the DOM 1`] = `
<div
  className="app container-fluid"
>
  <nav
    className="theNavbar navbar navbar-expand-sm navbar-light"
  >
    <a
      className="navbar-brand"
      href="/"
    >
      <img
        alt="Car Rentals"
        className="logo"
        src="logo.PNG"
      />
    </a>
    <button
      aria-controls="responsive-navbar-nav"
      aria-label="Toggle navigation"
      className="navbar-toggler collapsed"
      onClick={[Function]}
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      aria-expanded={null}
      className="navbar-collapse collapse"
      id="responsive-navbar-nav"
    >
      <div
        className="mr-auto navbar-nav"
        onKeyDown={[Function]}
      />
      <div
        className="d-flex justify-content-end navbar-nav"
        onKeyDown={[Function]}
      >
        <a
          className="nav-link white"
          href="/"
        >
          Home
        </a>
        <a
          className="nav-link white"
          href="/suits"
        >
          Suits
        </a>
        <a
          className="nav-link white"
          href="/appointments"
        >
          Appointments
        </a>
        <a
          className="nav-link white"
          href="/logout"
        >
          LOGOUT
        </a>
      </div>
    </div>
  </nav>
  <div
    className="homepage p-0 m-0 row"
  >
    <div
      className="banner col-12"
    >
      <div
        className="text-center"
      >
        <h1
          className="my-3"
        >
          The Gentleman
        </h1>
        <p>
          Character. Intelligence. Strength. Style
        </p>
        <p>
          That makes a gentleman.
        </p>
      </div>
    </div>
    <div
      className="welcomeText col"
    >
      <button
        className="my-2 btn btn-md btn-default border"
        type="button"
      >
        Explore More
      </button>
      <p
        className="m-2"
      >
        Explore the style, the elegant look for your special occassion. We offer high quility products. The gentleman is here to give you a sophisticated appearance and lighten up the mood of your day. Enjoy!
      </p>
    </div>
  </div>
  <div
    className="mt-2 pt-4 text-center footerContainer"
  >
    <p>
      Developed by Clayton Siby. Â©copyright 2021. All rights reserved.
    </p>
    <p>
      <span
        className="socialIcon"
      >
        <a
          href="https://github.com/ClaytonSiby"
          rel="noreferrer"
          target="_blank"
        >
          <i
            className="fab fa-github"
          />
        </a>
      </span>
      <span
        className="socialIcon"
      >
        <a
          href="https://www.linkedin/in/clayton-siby"
          rel="noreferrer"
          target="_blank"
        >
          <i
            className="fab fa-linkedin-in"
          />
        </a>
      </span>
      <span
        className="socialIcon"
      >
        <a
          href="https://twitter.com/ClaytonSiby"
          rel="noreferrer"
          target="_blank"
        >
          <i
            className="fab fa-twitter"
          />
        </a>
      </span>
    </p>
  </div>
</div>
`;
